File: 프로그래머스/lv1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.cs
using System;

public class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        int[] answer = new int[] {0,0};
        int rightNum=0;
        int empty=0;
        for(int i=0;i<lottos.Length;++i)
        {
            if(Array.Find(win_nums,(x) => x == lottos[i]) != 0)
                rightNum++;
            if(lottos[i] == 0)
                empty++;
        }
            
        answer[0] = Math.Clamp(6-(rightNum+empty)+1,1,6);
        answer[1] = Math.Clamp(6-rightNum+1,1,6);
       
        
        return answer;
    }
}

File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.cpp
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> lottos, vector<int> win_nums) {
    vector<int> answer;
    int count{};
    int unknownCount{};
    int lottosSize = lottos.size();
    int min{};
    int max{};
    
    for(int i=0;i<lottosSize;++i)
    {
        auto itr = find(win_nums.begin(),win_nums.end(),lottos[i]);
        if(itr != win_nums.end())
            ++count;
        if(lottos[i] == 0)
            ++unknownCount;
    }
    
    
    max = 7 - (count + unknownCount);
    if(max == 7)
        max = 6;
    answer.emplace_back(max);
    
    min = 7 - count;
    if(min == 7)
        min = 6;
    answer.emplace_back(min);
    
    return answer;
}